provider/gce/environ_broker.go

```diff
@@ -34,20 +34,32 @@ func (env *environ) StartInstance(args environs.StartInstanceParams) (*environs.
 
 	spec, err := buildInstanceSpec(env, args)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: true,
+		}
 	}
 
 	if err := env.finishInstanceConfig(args, spec); err != nil {
-		return nil, errors.Trace(err)
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: true,
+		}
 	}
 
 	// Validate availability zone.
 	volumeAttachmentsZone, err := volumeAttachmentsZone(args.VolumeAttachments)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: true,
+		}
 	}
 	if err := validateAvailabilityZoneConsistency(args.AvailabilityZone, volumeAttachmentsZone); err != nil {
-		return nil, errors.Wrap(errors.Trace(err), environs.ErrAvailabilityZoneFailed)
+		return nil, &common.StartInstanceError{
```

Comment:
Should these actually be a Function because then we get proper Trace behavior?

Reply:
I don't think so. That would lead to source context being added wherever the common function is, which isn't helpful. Tracing should be done at the point where it's returned. Also, adding a function would make this even more boilerplatey.